src/AbstractCsv.php

```diff
@@ -127,6 +128,35 @@ public static function createFromFileObject(SplFileObject $file)
     }
 
     /**
+     * Return a new {@link AbstractCsv} from a PHP resource stream
+     *
+     * @param resource $stream
+     *
+     * @return static
+     */
+    public static function createFromStream($stream)
+    {
+        if (is_resource($stream)) {
+            $stream = new StreamIterator($stream);
+        }
+
+        if ($stream instanceof StreamIterator) {
```

Comment:
well after review .. I need this check since the method can take a resource or a StreamIterator and I don't want to create 2 named constructors for this otherwise you could create a CSV object from a path 👎